lib/token.h

```diff
@@ -187,6 +187,30 @@ class CPPCHECKLIB Token {
     explicit Token(TokensFrontBack *tokensFrontBack = nullptr);
     ~Token();
 
+    struct TokenEnumeration
+    {
+        Token const* mt;
+        TokenEnumeration(Token const* t) : mt(t) {}
+
+        struct TokenIterator
+        {
+            const Token* mt;
+            TokenIterator(const Token* t) : mt(t) {}
+            TokenIterator& operator++() { mt = mt->next(); return *this; }
+            bool operator==(const TokenIterator& b) { return mt == b.mt; }
+            bool operator!=(const TokenIterator& b) { return mt != b.mt; }
+            const Token* operator*() { return mt; }
+        };
+
+        TokenIterator begin() { return TokenIterator(mt); }
+        TokenIterator end() { return TokenIterator(mt->mTokensFrontBack->back); }
+    };
+
+    TokenEnumeration toend() const
+    {
+        return TokenEnumeration(this);
+    }
```

Comment:
I repeatedly found myself wanting to iterate over tokens using range based for so I whipped something up. I'm not married to it and it's not required for this fix but I think it's a nice change.

Reply:
hmm.. I basically like the idea. I suggest we take it in a separate PR. I think we would want a mechanism that allows us to iterate until the end of a scope. Also I am not sure that token.h is the right place.

Reply:
Will address in #3056